src/layers/XRLayer/XRLayer.js

```diff
@@ -45,18 +51,18 @@ export default class XRLayer extends Layer {
   getShaders() {
     const { gl } = this.context;
     const noWebGL2 = !isWebGL2(gl);
-    const { colormap, dtype } = this.props;
+    const { colormap } = this.props;
     const fragShaderNoColormap = noWebGL2 ? fs1 : fs2;
     const fragShaderColoramp = noWebGL2 ? fsColormap1 : fsColormap2;
-    const fragShader = colormap
-      ? fragShaderColoramp.replace('colormapFunction', colormap)
-      : fragShaderNoColormap;
-    const fragShaderDtype =
-      dtype === '<f4' ? fragShader.replace(/usampler/g, 'sampler') : fragShader;
+    const fragShader = colormap ? fragShaderColoramp : fragShaderNoColormap;
     return super.getShaders({
       vs: noWebGL2 ? vs1 : vs2,
-      fs: fragShaderDtype,
-      modules: [project32, picking]
+      fs: fragShader,
+      defines: {
+        SAMPLER_TYPE: getSamplerType(this.props, noWebGL2),
+        COLORMAP_FUNCTION: colormap || 'viridis'
+      },
+      modules: [project32, picking, channels, lens]
     });
   }
```

Comment:
You could even move the `getShaderProps` code to this block, I just think organizing the imports in a data structure (e.g. `SHADER_MODULES`) makes it much easier to read.

```suggestion
  getShaders() {
    const shaderProps = getShaderProps(this.props, this.context.gl);
    return super.getShaders(shaderProps);
  }
```